### PR TITLE
Beta update dependencies babel - Initial babel upgrade (step 1)

### PR DESCRIPTION
**Pull request by pradeepnschrodinger**
## Description

Upgrade babel to use v7
ran `npx babel-upgrade --write` (see [babel-upgrade](https://github.com/babel/babel-upgrade))

## Navigate
#450 - Initial babel upgrade (*)
#451 - Calypso codemods
#452 - Fix site and tests
#453 - Fix internal build

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,9 +1,13 @@
 {
   "plugins": [
-    "transform-object-rest-spread",
-    "transform-object-assign",
-    "transform-class-properties",
+    ["@babel/plugin-transform-flow-strip-types", {"requireDirective": true}],
```
Comment (pradeepnschrodinger) [inline .babelrc]: Babel seems to barf up on files like PrefixIntervalTree.js, due to flow annotations.
Using this plugin in to strip them off.

**Review by Az111z2 — APPROVED** [review]

File: .babelrc (hunk 1 above)
Comment (Az111z2) [reply .babelrc:3]: A

Comment (Az111z2) [reply .babelrc:3]: > Babel seems to barf up on files like PrefixIntervalTree.js, due to flow annotations.
> Using this plugin in to strip them off.

Comment (Az111z2) [reply .babelrc:3]: A

**Pull request merged**